packages/visual-editor/src/editor/Editor.tsx

```diff
@@ -106,7 +107,9 @@ export const Editor = ({
 
   useEffect(() => {
     if (templateMetadata?.platformLocale) {
-      i18nPlatformInstance.changeLanguage(templateMetadata?.platformLocale);
+      i18nPlatformInstance.changeLanguage(
+        normalizeLocale(templateMetadata?.platformLocale)
```

Comment:
Normalize at the outermost level - so do this directly on the templateMetadata when we receive the message.